### DISCUSSION TITLE
Gate altitude calculation

### DISCUSSION THREAD

**Discussion by saranya00-s** [ARM-DOE/pyart#1657] [General]
Is the function pyart.core.Radar.get_gate_lat_lon_alt is returning earth curvature corrected heights? Please confirm.

**Comment by zssherman** [comment]
@saranya00-s that function returns the current lat lon and alt from a radar sweep. But those variables are calculated from:
https://github.com/ARM-DOE/pyart/blob/14f625a10633e92c12818c41ea9269322b25e1ee/pyart/core/transforms.py#L424
Which does calculate the arc length along Earth's curvature.

**Reply by saranya00-s** [reply]
I think this function "geographic_to_cartesian_aeqd(lon, lat, lon_0, lat_0, R=6370997.0)" returns x,y,z values. But here we need a antenna to geographic transformation. Then which coordinate transformation function is used in pyart.core.Radar.get_gate_lat_lon_alt ?

**Reply by zssherman** [reply]
Oops yeah correct, mistake on my part, I meant to share:
https://github.com/ARM-DOE/pyart/blob/14f625a10633e92c12818c41ea9269322b25e1ee/pyart/core/transforms.py#L599

**Reply by zssherman** [reply]
the x, y, z that get placed into that for retrieving the lat lon etc is from this:
https://github.com/ARM-DOE/pyart/blob/14f625a10633e92c12818c41ea9269322b25e1ee/pyart/core/transforms.py#L108

**Reply by zssherman** [reply]
So essentially ranges, elevations and azimuths into antenna_vectors_to_cartesian, then x, y, into cartesian_to_geographic_aeqd. Then the function for 'get' you shared above retrieve those lat lons for a specific sweep.

**Comment by saranya00-s** [comment]
Ok. Now it's clear. Thank you

**Reply by zssherman** [reply]
No problem!
